cmd/query/app/server_test.go

```diff
@@ -73,28 +73,37 @@ func TestServerBadHostPort(t *testing.T) {
 }
 
 func TestServerInUseHostPort(t *testing.T) {
+	const availableHostPort = "127.0.0.1:0"
+	conn, err := net.Listen("tcp", availableHostPort)
+	defer func() { conn.Close() }()
+	assert.NoError(t, err)
 
-	for _, hostPort := range [2]string{":8080", ":8081"} {
-		conn, err := net.Listen("tcp", hostPort)
-		assert.NoError(t, err)
-
-		server, err := NewServer(zap.NewNop(), &querysvc.QueryService{},
-			&QueryOptions{HTTPHostPort: "127.0.0.1:8080", GRPCHostPort: "127.0.0.1:8081", BearerTokenPropagation: true},
-			opentracing.NoopTracer{})
-		assert.NoError(t, err)
-
-		err = server.Start()
-		assert.NotNil(t, err)
-		conn.Close()
-		if server.grpcConn != nil {
-			server.grpcConn.Close()
-		}
-		if server.httpConn != nil {
-			server.httpConn.Close()
-		}
-
+	testCases := []struct {
+		name         string
+		httpHostPort string
+		grpcHostPort string
+	}{
+		{"HTTP host port clash on " + conn.Addr().String(), conn.Addr().String(), availableHostPort},
```

Comment:
`"HTTP host port clash on " + conn.Addr().String()` - I think it's better if the test names are stable, this one will include random port number on every run, which is not informative anyway